From ff6730b7d0a3e9139010b293822298d1da8456ab Mon Sep 17 00:00:00 2001
From: Sergey Dyasli <sergey.dyasli@citrix.com>
Date: Wed, 13 Mar 2019 16:26:48 +0000
Subject: [PATCH 1/8] microcode: fix ucode application on AMD

This was reported back to Chao and is expected to be included in v7.

Signed-off-by: Sergey Dyasli <sergey.dyasli@citrix.com>
diff --git a/xen/arch/x86/microcode_amd.c b/xen/arch/x86/microcode_amd.c
index a5e6d5f28f..7128146162 100644
--- a/xen/arch/x86/microcode_amd.c
+++ b/xen/arch/x86/microcode_amd.c
@@ -249,6 +249,7 @@ static enum microcode_match_result compare_patch(
 static int apply_microcode(void)
 {
     uint32_t rev;
+    const struct microcode_amd *mc_amd;
     const struct microcode_header_amd *hdr;
     const struct microcode_patch *patch;
     int hw_err;
@@ -259,7 +260,8 @@ static int apply_microcode(void)
     if ( patch == NULL )
         return -EINVAL;
 
-    hdr = patch->data;
+    mc_amd = patch->data;
+    hdr = mc_amd->mpb;
 
     BUG_ON(local_irq_is_enabled());
 
